internal/cmds/cmds.go

```diff
@@ -412,21 +456,27 @@ func rehydrateMrSeqFilesForProblematicUpgrades(ctx *log.Context, h types.Handler
 	}
 
 	if isProblematicVersion {
-		ctx.Log("message", fmt.Sprintf("Rehydrating mrseq files deleted by from version '%s' using status files", oldExtensionVersion))
-		return doRehydrateMrSeqFilesForProblematicUpgrades(ctx, oldExtensionDirectory, newExtensionDirectory)
+		message := fmt.Sprintf("Rehydrating mrseq files deleted by from version '%s' using status files", oldExtensionVersion)
+		ctx.Log("message", message)
+		extensionEvents.LogInformationalEvent("rehydratemrseq", message)
+		return doRehydrateMrSeqFilesForProblematicUpgrades(ctx, oldExtensionDirectory, newExtensionDirectory, extensionEvents)
 	} else {
-		ctx.Log("message", fmt.Sprintf("Previous extension version '%s' does not require mrseq hydration", oldExtensionVersion))
+		message := fmt.Sprintf("Previous extension version '%s' does not require mrseq hydration", oldExtensionVersion)
+		ctx.Log("message", message)
+		extensionEvents.LogInformationalEvent("rehydratemrseq", message)
 	}
 
 	return nil
 }
 
-func doRehydrateMrSeqFilesForProblematicUpgrades(ctx *log.Context, oldExtensionDirectory string, newExtensionDirectory string) error {
+func doRehydrateMrSeqFilesForProblematicUpgrades(ctx *log.Context, oldExtensionDirectory string, newExtensionDirectory string, extensionEvents *extensionevents.ExtensionEventManager) error {
 	oldExtensionStatusDirectory := filepath.Join(oldExtensionDirectory, constants.StatusFileDirectory)
 
 	extensionStatusDirectoryFDRef, err := os.Open(oldExtensionStatusDirectory)
 	if err != nil {
-		return errors.Wrap(err, fmt.Sprintf("Failed to open status directory '%s'", oldExtensionStatusDirectory))
+		errMessage := fmt.Sprintf("Failed to open status directory '%s'", oldExtensionStatusDirectory)
```

Comment:
add the error message here (err.Error())

Reply:
added